recipe_scrapers/thecookingguy.py

```diff
@@ -0,0 +1,99 @@
+# mypy: allow-untyped-defs
+from ._abstract import AbstractScraper
+from ._exceptions import ElementNotFoundInHtml
+from ._grouping_utils import IngredientGroup
+from ._utils import get_yields, normalize_string
+
+
+class TheCookingGuy(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "thecookingguy.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def total_time(self):
+        return None
+
+    def yields(self):
+        return get_yields(self.soup.find("div", class_="text-block-7").get_text())
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = self.soup.find(
+            "div", class_="w-layout-vflex card-text-holder ingredients"
+        ).find_all("li")
+        ingredients_text = [
+            normalize_string(ingredient.get_text()) for ingredient in ingredients
+        ]
+        return ingredients_text
+
+    def ingredient_groups(self):
+        # find ingredients div
+        ingredients_div = self.soup.find(
+            "div", class_="w-layout-vflex card-text-holder ingredients"
+        )
+
+        if ingredients_div is None:
+            raise ElementNotFoundInHtml("Ingredients not found.")
+
+        # find all ingredient group purposes
+        ingredient_group_ps = ingredients_div.find_all("p")
```

Comment:
To me it would make more sense to find all the ingredient lists, then use [previous_sibling](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#find-previous-siblings-and-find-previous-sibling) to retrieve the purpose (if it exists).

This would mean you could remove the duplicate code for grouped/ungrouped.

Reply:
Sounds reasonable, I think in my head the logic is easier to understand by working with the grouped items and then the ungrouped items. Using previous_sibling would reduce the code though, so we can refactor it to that. Thanks for the comment!

Reply:
No worries, great work so far!